### ISSUE TITLE
PVR recordings issues with Omega

### ISSUE THREAD

**Issue by popcornmix** [xbmc/xbmc#23199]
### Describe the bug

I have PVR issues on Omega/ffmpeg 6.
Resuming recordings doesn't work (the elapsed time is garbage and audio/video skips).

I've diffed detailed logging and first difference that seems significant is:
```
ffmpeg[0x353d868X]: [mpegts] read_seek: [0 5439510000](tel:05439510000)
ffmpeg[0x353d868X]: [mpegts] gen_seek: [0 5439510000](tel:05439510000)
```
on ffmpeg 4 and:
```
ffmpeg[0x351cea50]: [mpegts] read_seek: 0 -9223372036800810000
ffmpeg[0x351cea50]: [mpegts] gen_seek: 0 -9223372036800810000
```
on ffmpeg 6 (presumably a no pts value).

That invalid timestamp seems to propagate through VideoPlayer. e.g.
```CDVDMsg::GENERAL_RESYNC(9223372036854365184.000000)```
and
```PrepareNextRender: frameOnScreen: 600001839.741000 renderPts: 600068506.412753 nextFramePts:9223372036854775808.000000 -> diff: -9223372036254707712.000000  render: false forceNext: true")```

### To Reproduce
I can reproduce with a ts file played as a recording through PVR Demo app.
[example file](https://drive.google.com/uc?id=1fKOWcRZhYbBxJifC7XO7fAJ4L10QPN7R&export=download)

If I play, seek to 10 minutes then stop, then try to play and resume, ffmpeg 4.3.2 is good. ffmpeg 6.0 is bad
It shows elapsed time as 577014:32:43/52.03 and it constantly skips with "large audio sync error").

It doesn't matter what hardware acceleration is used.
The failure occurs with software decode and egl render, and also with DRMPrime.

### Debuglog

[Working log file with ffmpeg 4](http://sprunge.us/ucezmf)
[Broken log file with ffmpeg 6](http://sprunge.us/5PWP0e)

### Screenshots 
Here are some links or screenshots to help explain the problem:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 21

The logs are from my own builds of LibreELEC on Pi4.
I also see the issue in LE nightly builds.

I don't have an exact timeframe when the issue started, but it's been present for a number of weeks.
Most likely coinciding with one of the ffmpeg version bumps.

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by Portisch** [comment]
As the duplicate is closed let's continue here. We are also doing this for "Hobby" for free, unpaid.
But such big project need to look further that maybe other projects depends on their work. They sub project can't be better then the head.

Quite now nothing got really improved, only step backwards. And this breaks also sub projects. Not only CoreELEC, also main projects like LibreELEC.

Kodi become more and more a private instead a community project. And this only can end bad for users...

**Comment by fritsch** [comment]
What do you suggest to find community that contributes code and fixes to make it more community driven again?

**Comment by ksooo** [comment]
Nothing improved? Have you seen all the commits for 21? All the things that were added? What’s wrong with you? As I said if you looking for something stable user Nexus. 

If you really want to help us, go and make your hands dirty. Learn c++, dive into Kodi code bar, contribute. That’s what I did.

**Comment by Portisch** [comment]
"Improvements" are under the hood and a end user will not see it. There is nothing different for them. Users aren't interested in changes under the hood, only in new features and visible changes.

I can handle c++, sure better ANSI c. I did dig 3 days into this issue as I first thought it's our project issue.
So I don't blame Kodi on first place. I would never without bulletproof testing where the issue is located. But then I realized it's on all platforms. I did post my findings and how you can replicate the issue. So you can start within 5min debugging. But the issue is closed now, because of duplicate...

On comments like this: my hair are raising to heaven:
https://github.com/xbmc/xbmc/commit/dc4c5dd28954dd5b8062f96818302ec86a023feb

Why replace working code with blocking one? This replacement do now cause deadlocks. Like when library scanning on SMB is active and Kodi want to exit.

Such loops with unlock, wait 1ms, lock again are worst multithreading handling I have seen ever. Sure it's historical code but it breaks and prevent a lot of new features. Like PiP is possible on every platform in hardware. But with this code base it's just not possible to implement such feature as it will break everything.
Like since Nexus the video thread is not finished and a new one is already started when play/play and not play/stop/play a file. So two decoders get opened in same time.

On the other side issues breaking essential handling uncommitted for months. This normally should light up ever warning light like fire is on the roof. But nothing.

Also PRs when they don't match the "private" interests for some persons are just ignored. Even if they really bring new features to Kodi:
https://github.com/xbmc/xbmc/pull/22546#issuecomment-1595782590

And TK should be have a bit more interests into subprojects as like we do use Kodi on a wide range of users on nightly base. So issues are triggered hopefully soonest as possible and do not bork up real releases.

**Comment by ksooo** [comment]
Your project pointing your users to your builds based on our nightlife’s which we claim unstabile alpha and then to blame us for bad quality and to request that we should keep alpha stable because of your use case is simply ridiculous.

**Comment by Portisch** [comment]
Sorry then for request quality we are used to in past Kodi versions. Knowing issues for months and do make 2 alphas without take a look at all into the issue complete the set if quality.

Thanks for your help, we will put Omega on ice than.

**Comment by ksooo** [comment]
Are you from another planet or what? Is it so hard to understand that we cannot force anybody to work on anything. Being a Team Kodi member does not mean any obligation to spend time on Kodi. If we currently do not have resources to look at all the issues or just lacking skills to deal with some of the issues, what do you suggest we should do about this? Making clear that alpha means potentially unstable and buggy should be more than sufficient.

**Comment by emveepee** [comment]
Not being on Team Kodi I can still appreciate the help they have given me, to help NextPVR users.  I, like many users primarily depend on CoreELEC and LibreELEC so I equally appreciate what Portisch, popcornhour and the CE and LE teams have done to make the PVR experience truly what it is.  PVR clients on dedicated boxes are  the way to go IMO so this is an incredibly important union of open source projects. 

I can say from my pvr.nextpvr perspective I am surprised that these extremely significant issues with mpeg-ts has not been triaged or acknowledged for so long, especially as you move through alpha stages.    Certainly pvr.nextpvr recordings are suffering the same issues that have been reported.  Personally I don't think it is ffmpeg 6 specific because the Snocker2 sample resumes fine on my libmpv based client.

I don't have the technical skills to delve into ffmpeg handling in Kodi but I can build test builds on any platform if I can be be of any assistance.   I am fortunate that my main client machine dual boots CE 21 and Android so it is not hard to have Nexus and Omega on the same box.

**Comment by fritsch** [comment]
Long shot: https://www.mail-archive.com/ffmpeg-trac@avcodec.org/msg61276.html removed an ancient hack for broken mpegts files, cause someone complained that opening them would take too long: 58df81b02779c9c87cbf5403cfdbb437d5344669 <- relevant ffmpeg commit.

Does that solve the issue at hand?

Would be a typical thing where stuff starts to break. As mentioned in the link above it is not a practical issue for ffmpeg or ffplay, cause those clients set some flags that proper reinitialization happens.

As said: long shot.

**Comment by rainman74** [comment]
> Are you from another planet or what? Is it so hard to understand that we cannot force anybody to work on anything. Being a Team Kodi member does not mean any obligation to spend time on Kodi. If we currently do not have resources to look at all the issues or just lacking skills to deal with some of the issues, what do you suggest we should do about this? Making clear that alpha means potentially unstable and buggy should be more than sufficient.

Briefly To the other thread: This was not meant to be bashing, sorry! The idea was just that you might pass it on to a colleague who is familiar with the VideoPlayer.

The problem affects not only MPEG-TS but also MKV files.
Skipping back and forth is no longer possible, which causes Kodi to hang or crash completely.

**Comment by ksooo** [comment]
To my understanding this is something ffmpeg6 related. Something I really have no knowledge about, sorry. I’m out at this point.

**Comment by rainman74** [comment]
> fritsch

Is there a test build?

**Comment by fritsch** [comment]
No - I just asked this question to the folks that already looked some days into it. I simply checked the mpegts changes of ffmpeg6 to see if that might fit. I hope the people who already dived into that issue might have more information.

**Comment by popcornmix** [comment]
@fritsch I tried reverting that ffmpeg commit, but unfortunately issue is still present.

**Comment by fritsch** [comment]
Was worth a try. If we all think that it is an ffmpeg regresion (in combination with kodi), meaning: Current kodi code works fine with ffmpeg5.x - then bisecting ffmpeg 6 would be one way to go.

If we think the changes on kodi side, that were introduced together with ffmpeg6 are the issue, then bisecting ffmpeg is not a solution.

**Comment by fritsch** [comment]
If it's on kodi side: https://github.com/xbmc/xbmc/pull/22967/commits/c12af890b0973f7c86316087e823f8a31c6b2ed3 <- the reordered_opaque that we used before is definitely a candidate.

The user sets [AVCodecContext.reordered_opaque](https://ffmpeg.org/doxygen/0.9/structAVCodecContext.html#64c09907902b41744c3ea60c89fd42fb) to represent the input at that time, the decoder reorders values as needed and sets [AVFrame.reordered_opaque](https://ffmpeg.org/doxygen/0.9/structAVFrame.html#b54a78c51318c8c678e545b1f3d6fddd) to exactly one of the values provided by the user through [AVCodecContext.reordered_opaque](https://ffmpeg.org/doxygen/0.9/structAVCodecContext.html#64c09907902b41744c3ea60c89fd42fb)

Especially seeing your negative vs. positive time stamps looks like it could cause that issue.

**Comment by fritsch** [comment]
This should be easily testable: reordered_opaque is still there in ffmpeg - so changing the macro in the pointed out file to > ffmpeg version 7 will make that run again with the reordered_opaque plus some deprecation warnings obviously.

**Comment by rainman74** [comment]
As far as the problem with the skipped forward and backward playback is concerned, I can say that this still worked in the release alpha2 of 17.06., but then no longer in the further nightly on 23.06.

But of course it could also be a different problem.

What I can also say is that between Alpha1 and Alpha2, the recognition of the end of the video was also different. Since Alpha2 it now takes up to 30 seconds until the next video in the playlist starts. Otherwise it always started immediately when the end was reached. But this may also be another separate problem.

**Comment by fritsch** [comment]
The issue filed here is from April 2023.

**Comment by popcornmix** [comment]
I tried reverting https://github.com/xbmc/xbmc/commit/c12af890b0973f7c86316087e823f8a31c6b2ed3 but it didn't fix the issue.

**Comment by fritsch** [comment]
Just to be fully sure: ffmpeg5 works fine, right? You only stated above that ffmpeg4 is fine. So might very well be that this regression already slipped in when we moved to ffmpeg5.

**Comment by fritsch** [comment]
So - if the regression was already there after our ffmpeg5 bump: https://github.com/xbmc/xbmc/pull/21248/commits/0a10e05700dfd12b7f5d720588b993089ea33be4 <- is a candidate as we already had mpegts issues there but manually solved it.

**Comment by emveepee** [comment]
I can see here on resume play that m.pkt.pkt.dts is invalid causing a crazy m_startTime here https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L2407

![image](https://github.com/xbmc/xbmc/assets/2148031/36448552-2638-4e3f-8a5f-3ac98d8536dc)

**Comment by fritsch** [comment]
You see in your screenshot, that the dts is roughly the value @popcornmix showed above for the "working ffmpeg4" in correlation with his seek, while the new version shows an unsigned value.

So wondering if we overwrite our pts somewhere with the broken dts value.

**Comment by emveepee** [comment]
I didn't try with v5 because @Portisch reported it wasn't working either https://github.com/xbmc/xbmc/issues/23431#issuecomment-1607063341

**Comment by fritsch** [comment]
Good - then I am quite sure we broke that already with ffmpeg5 months ago, but no one realized. Then I can stop looking into ffmpeg6 regressions, as it is a ffmpeg5 / kodi ffmpeg5 regressions, e.g. the commit I pointed out above.

**Comment by popcornmix** [comment]
_can confirm issue occurs with oldest LE nightly available (LibreELEC-RPi4.aarch64-12.0-nightly-20230310-91357b0) which uses ffmpeg 5.1.2. But there's no older builds to easily test.

**Comment by fritsch** [comment]
Thanks very much. This is what I expected. After the initial commit mpegts and especially tvheadend did not work at all, even after the audio channel layout was initialized, which is why the change I pointed out above was implemented to get mpegts and tvh working. It was tested by some tvh users, but it seems this scenario above was forgotten.

**Comment by emveepee** [comment]
When I a compare working Nexus with Omega the change from the removed st->cur_dts to m_pkt.pkt.dts https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L2408 may not be a simple replacement.  

Although not the same thing after changing that line to use st->start_time resume on my recordings do work.

**Comment by fritsch** [comment]
Maybe we can ask the original author: https://github.com/FFmpeg/FFmpeg/commit/591b88e6787c4e678237f02a50421d101abd25c2

**Comment by fritsch** [comment]
Others have the same problem: https://ffmpeg.org/pipermail/ffmpeg-devel/2021-September/285401.html

**Comment by fritsch** [comment]
And here is how he fixed that from the cli tools: https://github.com/FFmpeg/FFmpeg/commit/ab4f299e23

**Comment by fritsch** [comment]
Your start_time goes into the right direction, the fix above remembers the first_dts as well.

**Comment by emveepee** [comment]
I thought this comment and the ones that precede it might be relevant to this discussion https://github.com/xbmc/xbmc/pull/5661#issuecomment-62105682 since the v5 changes do use the packet based dts.

**Comment by rainman74** [comment]
> The issue filed here is from April 2023.

Then perhaps the other bug report should not simply have been closed without knowing what the reason is, one simply points to a *suspected* duplicate.

In the other thread, the forwarding and rewinding was reported in detail and also by several users with Android TV and also MKV files.

This one:
https://github.com/xbmc/xbmc/issues/23431

**Comment by emveepee** [comment]
I think all the changes for st->cur_pts need to be investigated not just the one I highlighted.  Additionally just skipping when m_pkt.pkt.dts is AV_NOPTS_VALUE (which incidentally is -9223372036854775808 ) isn't good enough.

**Comment by popcornmix** [comment]
Just to confirm what has been discussed, I tried adjusting the 5 instances of `m_pkt.pkt.dts` that previously were `st->cur_dts` back to cur_dts.
You can get this using sti->cur_dts where `const FFStream *const sti = cffstream(st);` using FFStream/cffstream from internal.h as ffmpeg still maintains this value internally.
Obviously this is just a hack to confirm this is the issue, but I can confirm it fixes the issue I reported.

I guess the right fix is to calculate `cur_dts` inside DVDDemuxFFMPEG using the same logic ffmpeg does, and use that value in the 5 locations.

**Comment by emveepee** [comment]
The change in UpdateCurrentPTS() should probably be using st->start_time anyway since it is referring to pts not dts.

**Comment by popcornmix** [comment]
Unfortunately I suspect that calculating `cur_dts` externally will be impossible.
I had assumed it was a fairly simple combination of previous `dts` and `pts` values. 
But looking more closely it's very convoluted, and there are both places where
`cur_dts` is assigned to a packet `dts`/`pts` and cases where a packet `dts`/`pts` is assigned to `cur_dts`
I suspect the information is not visible externally to produce an exact match.

So the question is whether there exist alternative values that will suffice in the places we had been using `cur_dts`.
The code we currently have is clearly wrong.
For example after a seek, we call `UpdateCurrentPTS` which uses `m_pkt.pkt.dts`
But we've just seeked and haven't read a packet so that is just a stale value we are using.

Possibly after the seek using `m_pFormatContext->start_time` is more appropriate.
Or maybe `av_find_default_stream_index(m_pFormatContext)->start_time`.
However neither exactly matches the value you got in the old `cur_dts` version.
I'll have to see if it causes any issues.

**Comment by emveepee** [comment]
True pts and dts are not exactly the same, I guess what needs to be looked at is Kodi ever uses the delta.  The fact that GetCurrentPTS() is actually using dts in v4, and in v5/6  makes me doubt that it does.   Samples from users saying that mkv files do not skip (not just resume) in Omega probably need to be made available though to help explain why they aren't working.

**Comment by ksooo** [comment]
Maybe @FernetMenta has an idea how to fix this.

**Comment by FernetMenta** [comment]
@popcornmix is right, calling UpdateCurrentPTS after a seek produces garbage now. I don't understand how #21248 could have ever be merged. Why do people change code without trying to figuere how this might impact behaviour? 
One the other hand, @popcornmix , UpdateCurrentPTS should not have any impact on the issue discussed here because m_currentPTS is only used for chapters (as fas I could see after a brief look into the code). 
@emveepee the delta of pts and dts is not important here. At demuxer stage dts is monotonic but pts is not (packets get reordered after decoding)
My guess is that it nevers gets out of this loop https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L1292 because m_pkt.pkt.dts is never set.

**Comment by FernetMenta** [comment]
If you don't get the timestamps right after a seek anymore, you might need to read out some packets to set dts.

**Comment by FernetMenta** [comment]
@popcornmix can you check if m_pkt.pkt belongs to the stream st-> ? https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L2408
This is not necessarily the case for the current code. Maybe this if statement there (and a couple of others) https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L2401 should also check it the read packet belongs to the stream.

**Comment by popcornmix** [comment]
The original log files have expired (and I can't find them locally).

I've captured new ones. With this I play a recording, seek to 10m, stop, then resume.

With LibreELEC 11 (ffmpeg 4)
https://paste.ubuntu.com/p/T6XbFd8Kmj/

With LibreELEC 12  (ffmpeg 6)
https://paste.ubuntu.com/p/HVp54nC85D/

I've got ffmpeg + timestamp logging on so the logs are quite large (despite only playing for a second or so).

**Comment by popcornmix** [comment]
> can you check if m_pkt.pkt belongs to the stream st-> 

just to confirm, do you want to check `m_pkt.pkt.stream_index == idx`? (or `i` when `m_program == UINT_MAX`).
(I'm not sure if this is also equal to `st->index`).

**Comment by FernetMenta** [comment]
> m_pkt.pkt.stream_index == idx

Yes, I think this should be true if the packet belongs to the stream.

**Comment by popcornmix** [comment]
@rainman74 I've had a quick play, and for normal local files I've tried:
playing normally until file hits end
Switch to ff x32 until they file hits end
Seek multiple times until file hits end

each time the file exits (and if a playlist is set up it moves to next item).

I've also tried with a PVR recording and that also played okay.

Can you give more details on what you are playing (i.e. local file, PVR recording, internet stream etc),
what the file container format is (e.g. mkv, ts etc), the codec (h264 etc) and exactly how you are skipping
(e.g. which keypresses).
What platform are you on? Any different with HW decode disabled?

Are you seeing the issue with [this URL](http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4)

A debug log may also be useful.

**Comment by enen92** [comment]
I think this is causing regressions for me too, all my Pvr channels (hdhomerun) only have audio now:
https://paste.kodi.tv/yekubaxazu

Vs before:
https://paste.kodi.tv/izopugadel

**Comment by rainman74** [comment]
> @rainman74 I've had a quick play, and for normal local files I've tried: playing normally until file hits end Switch to ff x32 until they file hits end Seek multiple times until file hits end
> 
> each time the file exits (and if a playlist is set up it moves to next item).
> 
> I've also tried with a PVR recording and that also played okay.
> 
> Can you give more details on what you are playing (i.e. local file, PVR recording, internet stream etc), what the file container format is (e.g. mkv, ts etc), the codec (h264 etc) and exactly how you are skipping (e.g. which keypresses). What platform are you on? Any different with HW decode disabled?
> 
> Are you seeing the issue with [this URL](http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4)
> 
> A debug log may also be useful.

The problem occurs primarily with MP4 (h.264) files on Android that are stacked. So video_part1.mp4, video_part2.mp4, video_part3.mp4,...

As soon as you fast skip several seconds or minutes at a time, so that you go beyond the end of the current part, the player hangs and the video stops and the next part is no longer played.

I will add a debug log later...

**Comment by popcornmix** [comment]
Can you confirm that this works with master branch just before this PR was merged?

**Comment by FernetMenta** [comment]
@popcornmix this stacking use case looks very similar to the chapter one. I think it requires setting currentPts after a seek.

**Comment by popcornmix** [comment]
Setting currentPts to what? Does this require reading (potentially multiple) packets to find a valid timestamp?
Are we happy if those packets are lost?

Or is setting it to, say, the requested seek time close enough?

**Comment by FernetMenta** [comment]
> Setting currentPts to what? Does this require reading (potentially multiple) packets to find a valid timestamp?

yes

> Are we happy if those packets are lost?

Yes. If it doesn't have a timestamp, it won't be worth anything after a seek

>Or is setting it to, say, the requested seek time close enough?

Probably not because seek might be off too much.

**Comment by popcornmix** [comment]
Are you happy with the timestamp from a packet from any stream?
Or do you think it needs to be the timestamp from the "main" stream (e.g. video if hasVideo is true, else audio)?

**Comment by FernetMenta** [comment]
> Are you happy with the timestamp from a packet from any stream? Or do you think it needs to be the timestamp from the "main" stream (e.g. video if hasVideo is true, else audio)?

I think it should come from m_seekStream it you do not intend to dig into the details of all possible streams of mpegts and how much dts can be off related to seekStream. Setting it to seekStream would reinstate the old behaviour.

**Comment by FernetMenta** [comment]
@popcornmix make sure that the first packet read with a timestamp isn't thrown away.

**Comment by popcornmix** [comment]
@rainman74 does https://github.com/xbmc/xbmc/pull/23629 help your issue?

**Comment by fuzzard** [comment]
@rainman74 what platform do you need it built for? We really need this tested before it is merged in case further changes are required.

**Comment by rainman74** [comment]
> @rainman74 what platform do you need it built for? We really need this tested before it is merged in case further changes are required.

armeabi-v7a

**Comment by fuzzard** [comment]
https://jenkins.kodi.tv/job/android-arm-docker/3254/artifact/kodi-20230817-dd42563f-PR23629-armeabi-v7a.apk

Should be available there for another day or two.

**Comment by rainman74** [comment]
> kodi-20230817-dd42563f-PR23629-armeabi-v7a.apk

I installed the test app and unfortunately did not notice any change.

Therefore here is the debug log for it.

Steps:
- Started the video file
- Then fast forwarded to the next part and the video hangs
- Then I waited a bit and stopped the hanging video and closed the app

kodi.log:
https://drive.google.com/file/d/1MR_CmOHk7vyqCWiF3mFwJOnxS1betMbn/view?usp=sharing

**Comment by popcornmix** [comment]
Yours may be a different issue.

On a Pi4 I've tested a stacked video file (i.e. a file.part1.avi / file.part2.avi) and kodi correctly played past the join point. Seeking forwards and backwards across it also worked. As did fast forward.

**Comment by rainman74** [comment]
> https://jenkins.kodi.tv/job/android-arm-docker/3254/artifact/kodi-20230817-dd42563f-PR23629-armeabi-v7a.apk
> 
> Should be available there for another day or two.

I just tested it again with the latest nightly. It only doesn't work if the part files are in MP4 format. I converted them to MKV (only the container) and then the fast forward to the next part works again!

Does not work:
video part1.mp4
video part3.mp4
video part4.mp4


Works:
video part1.mkv
video part3.mkv
video part4.mkv

**Comment by popcornmix** [comment]
And does this work with an older nightly? (e.g. before https://github.com/xbmc/xbmc/pull/23557)?

**Comment by rainman74** [comment]
> #23557

I can't pinpoint the nightly build and commit, but yes, it was working before all these changes in this context.

So I can't 100% confirm #23557 as the cause of the error.

The last release what was bug free for this was kodi-21.0-Omega_alpha1-armeabi-v7a.apk, after that the fast skipping with mp4 part files broke some nightly build. I'm sorry I didn't make a more detailed note of this.

**Comment by rainman74** [comment]
> And does this work with an older nightly? (e.g. before #23557)?

No idea what was done there now, but almost skip does not work now with the current nightly build even in a current part file! It takes minutes to jump to the first skip position within the part (mp4/h.264). Should I create a new issue?

**Comment by popcornmix** [comment]
Yes, unless you can confirm that the issue started with the first build after https://github.com/xbmc/xbmc/pull/23557 then it's not clear why this issue is considered related.

The fact that this issue is called "PVR recording issues with Omega" and you issue has nothing to do with PVR does suggest a new issue would be best.

**Comment by rainman74** [comment]
> Yes, unless you can confirm that the issue started with the first build after #23557 then it's not clear why this issue is considered related.
> 
> The fact that this issue is called "PVR recording issues with Omega" and you issue has nothing to do with PVR does suggest a new issue would be best.

I have created a new issue for this and described it in detail:
https://github.com/xbmc/xbmc/issues/23673
